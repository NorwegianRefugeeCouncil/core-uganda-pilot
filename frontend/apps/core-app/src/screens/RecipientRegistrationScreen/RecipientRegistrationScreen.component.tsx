import * as React from 'react';
import { ScrollView as SV } from 'react-native';
import { Button, HStack, ScrollView, VStack } from 'native-base';
import { useForm, FormProvider, FieldValue } from 'react-hook-form';
import { FormWithRecord } from 'core-api-client';
import { Recipient } from 'core-api-client/src/types/client/Recipient';

import * as ReactHookFormTransformer from '../../utils/ReactHookFormTransformer';
import { RecipientEditor } from '../../components/Recipient/RecipientEditor';
import { RecipientViewer } from '../../components/Recipient/RecipientViewer';

type Props = {
  mode: 'edit' | 'review';
  data: FormWithRecord<Recipient>[];
  onSubmit: (data: FormWithRecord<Recipient>[]) => void;
  onCancel: () => void;
  error: string | null;
  loading: boolean;
};

export const RecipientRegistrationScreenComponent: React.FC<Props> = ({
  mode,
  data,
  onSubmit,
  onCancel,
  error,
  loading,
}) => {
  const scrollRef = React.useRef<SV | null>(null);

  React.useEffect(() => {
    scrollRef.current?.scrollTo({
      y: 0,
      animated: true,
    });
  }, [mode]);

  const f = useForm({
    mode: 'all',
    defaultValues: ReactHookFormTransformer.toReactHookForm(data),
  });

  React.useEffect(() => {
    f.reset(ReactHookFormTransformer.toReactHookForm(data));
  }, [JSON.stringify(data)]);

  const handleSubmit =
    mode === 'edit'
      ? f.handleSubmit((submittedData: FieldValue<any>) => {
          onSubmit(
            ReactHookFormTransformer.fromReactHookForm(data, submittedData),
          );
        })
      : () => onSubmit(data);

  if (
    loading ||
    error ||
    data.length === 0 ||
    Object.keys(f.getValues()).length === 0
  )
    return null;

  return (
    <FormProvider {...f}>
      <ScrollView ref={scrollRef} width="100%" maxWidth="1180px" marginX="auto">
        <VStack space={4}>
          {mode === 'edit' && <RecipientEditor data={data} />}
          {mode === 'review' && <RecipientViewer data={data} />}
          <HStack space={4} justifyContent="flex-end">
            <Button
              key={`cancel-${mode}`}
              testID="recipient-registration-cancel-button"
              onPress={onCancel}
              colorScheme="secondary"
              variant="minor"
            >
              {mode === 'edit' ? 'Cancel' : 'Back'}
            </Button>
            <Button
              key={`save-${mode}`}
              testID="recipient-registration-submit-button"
              onPress={handleSubmit}
              colorScheme="primary"
              variant="major"
            >
              {mode === 'edit' ? 'Review' : 'Save'}
            </Button>
          </HStack>
        </VStack>
      </ScrollView>
    </FormProvider>
  );
};
