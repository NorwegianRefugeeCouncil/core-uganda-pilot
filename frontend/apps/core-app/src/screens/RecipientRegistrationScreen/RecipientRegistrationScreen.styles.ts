import styled from 'styled-components/native';
import { Box } from 'native-base';

export const Container = styled(Box)``;
