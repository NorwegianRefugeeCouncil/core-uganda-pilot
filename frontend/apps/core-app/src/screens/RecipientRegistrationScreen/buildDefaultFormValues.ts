import { FormDefinition, Record } from 'core-api-client';

export const buildDefaultFormValues = (
  forms: FormDefinition[],
  records: Record[],
): {
  [formId: string]: {
    [fieldId: string]: any;
  };
} =>
  forms.reduce((acc, form) => {
    const record: Record | undefined = records.find(
      (r) => r.formId === form.id,
    );
    if (!record) return acc;
    return {
      ...acc,
      [form.id]: record.values.reduce((innerAcc, fieldValue) => {
        return {
          ...innerAcc,
          [fieldValue.fieldId]: fieldValue.value,
        };
      }, {}),
    };
  }, {});
