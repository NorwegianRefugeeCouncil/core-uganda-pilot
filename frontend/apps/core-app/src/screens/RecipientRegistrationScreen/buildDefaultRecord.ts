import {
  FieldKind,
  FormDefinition,
  getFieldKind,
  Record,
} from 'core-api-client';

export const buildDefaultRecord = (form: FormDefinition): Record => ({
  id: '', // We are creating records so there shouldn't be an id
  databaseId: form.databaseId,
  formId: form.id,
  ownerId: undefined,
  values: form.fields.map((field) => {
    const fieldType = getFieldKind(field.fieldType);
    switch (fieldType) {
      case FieldKind.Text:
      case FieldKind.MultilineText:
        return {
          fieldId: field.id,
          value: '',
        };
      case FieldKind.Reference:
      case FieldKind.Date:
      case FieldKind.Month:
      case FieldKind.Week:
      case FieldKind.SingleSelect:
        return {
          fieldId: field.id,
          value: null,
        };
      case FieldKind.MultiSelect:
        return {
          fieldId: field.id,
          value: [],
        };
      case FieldKind.Checkbox:
        return {
          fieldId: field.id,
          value: 'false',
        };
      case FieldKind.SubForm:
        return {
          fieldId: field.id,
          value: [],
        };
      default:
        return {
          fieldId: field.id,
          value: '',
        };
    }
  }),
});
