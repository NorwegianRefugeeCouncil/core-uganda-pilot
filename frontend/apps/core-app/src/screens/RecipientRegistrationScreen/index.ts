export { RecipientRegistrationScreenComponent as RecipientRegistrationScreen } from './RecipientRegistrationScreen.component';
