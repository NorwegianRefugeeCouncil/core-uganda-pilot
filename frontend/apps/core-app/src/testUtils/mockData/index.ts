export { makeRecord } from './Record';
export { makeForm } from './Form';
export { makeField } from './FormField';
