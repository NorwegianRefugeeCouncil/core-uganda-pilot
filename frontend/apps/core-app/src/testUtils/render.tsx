import * as React from 'react';
import { NavigationContainer } from '@react-navigation/native';
import {
  render as r,
  RenderAPI,
  RenderOptions,
} from '@testing-library/react-native';
import { render as rWeb } from '@testing-library/react';
import { NativeBaseProvider } from 'native-base';
import { theme } from 'core-design-system';

const inset = {
  frame: { x: 0, y: 0, width: 0, height: 0 },
  insets: { top: 0, left: 0, right: 0, bottom: 0 },
};

export const render = (
  component: React.ReactElement<any>,
  options?: RenderOptions | undefined,
): RenderAPI =>
  r(
    <NativeBaseProvider theme={theme} initialWindowMetrics={inset}>
      <NavigationContainer>{component}</NavigationContainer>
    </NativeBaseProvider>,
    options,
  );

export const renderWeb = (
  component: React.ReactElement<any>,
  options?: RenderOptions | undefined,
) =>
  rWeb(
    <NativeBaseProvider theme={theme} initialWindowMetrics={inset}>
      <NavigationContainer>{component}</NavigationContainer>
    </NativeBaseProvider>,
    options,
  );
