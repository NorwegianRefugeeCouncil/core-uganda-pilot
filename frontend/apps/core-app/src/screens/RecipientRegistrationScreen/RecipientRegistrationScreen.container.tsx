import * as React from 'react';
import { FormWithRecord } from 'core-api-client';
import { Recipient } from 'core-api-client/src/types/client/Recipient';
import { StackScreenProps } from '@react-navigation/stack';

import { formsClient } from '../../clients/formsClient';
import { useAPICall } from '../../hooks/useAPICall';
import { routes } from '../../constants/routes';
import { RootNavigatorParamList } from '../../navigation/root';

import { RecipientRegistrationScreenComponent } from './RecipientRegistrationScreen.component';

type Props = StackScreenProps<RootNavigatorParamList, 'recipientsRegistration'>;

export const RecipientRegistrationScreenContainer: React.FC<Props> = ({
  route,
  navigation,
}) => {
  const [mode, setMode] = React.useState<'edit' | 'review'>('edit');
  const [data, setData] = React.useState<FormWithRecord<Recipient>[]>([]);

  const [_, getRecipientFormsState] = useAPICall(
    formsClient.Form.getAncestors,
    [route.params.formId],
    true,
  );

  const [saveRecipient, saveRecipientState] = useAPICall(
    formsClient.Recipient.create,
    [data],
    false,
  );

  React.useEffect(() => {
    const unsubscribe = navigation.addListener('blur', () => {
      setMode('edit');
      setData([]);
    });

    return unsubscribe;
  }, [navigation]);

  React.useEffect(() => {
    setData(
      (getRecipientFormsState.data || []).map((form) => ({
        form,
        record: formsClient.Record.buildDefaultRecord(form),
      })),
    );
  }, [JSON.stringify(getRecipientFormsState.data)]);

  React.useEffect(() => {
    if (saveRecipientState.data) {
      navigation.navigate(routes.recipientsProfile.name, {
        recordId:
          saveRecipientState.data[saveRecipientState.data.length - 1].record.id,
        formId: route.params.formId,
        databaseId: route.params.databaseId,
      });
    }
  }, [JSON.stringify(saveRecipientState.data)]);

  const handleSubmit = (d: FormWithRecord<Recipient>[]) => {
    if (mode === 'edit') {
      setData(d);
      setMode('review');
    }
    if (mode === 'review') {
      saveRecipient();
    }
  };

  const handleCancel = () => {
    if (mode === 'edit') {
      navigation.navigate(routes.recipientsList.name);
    }
    if (mode === 'review') {
      setMode('edit');
    }
  };

  return (
    <RecipientRegistrationScreenComponent
      mode={mode}
      data={data}
      onSubmit={handleSubmit}
      onCancel={handleCancel}
      error={getRecipientFormsState.error}
      loading={getRecipientFormsState.loading}
    />
  );
};
