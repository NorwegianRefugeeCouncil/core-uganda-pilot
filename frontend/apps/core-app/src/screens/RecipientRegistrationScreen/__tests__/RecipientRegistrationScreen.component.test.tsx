import { fireEvent, waitFor } from '@testing-library/react-native';
import { FormWithRecord } from 'core-api-client';
import { Recipient } from 'core-api-client/src/types/client/Recipient';

import { render } from '../../../testUtils/render';
import * as ReactHookFormTransformer from '../../../utils/ReactHookFormTransformer';
import { RecipientRegistrationScreenComponent } from '../RecipientRegistrationScreen.component';
import { makeFormWithRecord } from '../../../testUtils/mockData';

jest.mock('../../../components/Recipient/RecipientEditor', () => {
  const { View, Text } = jest.requireActual('react-native');
  return {
    RecipientEditor: ({ data }: { data: FormWithRecord<Recipient>[] }) => (
      <View>
        <Text>{JSON.stringify(data)}</Text>
      </View>
    ),
  };
});

jest.mock('../../../components/Recipient/RecipientViewer', () => {
  const { View, Text } = jest.requireActual('react-native');
  return {
    RecipientViewer: ({ data }: { data: FormWithRecord<Recipient>[] }) => (
      <View>
        <Text>{JSON.stringify(data)}</Text>
      </View>
    ),
  };
});

afterEach(() => {
  jest.clearAllMocks();
});

describe('should match the snapshot', () => {
  it('edit mode', () => {
    const { toJSON } = render(
      <RecipientRegistrationScreenComponent
        mode="edit"
        data={[
          makeFormWithRecord(1),
          makeFormWithRecord(2),
          makeFormWithRecord(3),
        ]}
        onSubmit={() => {}}
        onCancel={() => {}}
        error={null}
        loading={false}
      />,
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('review mode', () => {
    const { toJSON } = render(
      <RecipientRegistrationScreenComponent
        mode="review"
        data={[
          makeFormWithRecord(1),
          makeFormWithRecord(2),
          makeFormWithRecord(3),
        ]}
        onSubmit={() => {}}
        onCancel={() => {}}
        error={null}
        loading={false}
      />,
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('error', () => {
    const { toJSON } = render(
      <RecipientRegistrationScreenComponent
        mode="edit"
        data={[
          makeFormWithRecord(1),
          makeFormWithRecord(2),
          makeFormWithRecord(3),
        ]}
        onSubmit={() => {}}
        onCancel={() => {}}
        error="error"
        loading={false}
      />,
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('loading', () => {
    const { toJSON } = render(
      <RecipientRegistrationScreenComponent
        mode="edit"
        data={[
          makeFormWithRecord(1),
          makeFormWithRecord(2),
          makeFormWithRecord(3),
        ]}
        onSubmit={() => {}}
        onCancel={() => {}}
        error={null}
        loading
      />,
    );
    expect(toJSON()).toMatchSnapshot();
  });
});

describe('onSubmit', () => {
  it('should transform the form values, then submit', async () => {
    const fromReactHookFormSpy = jest.spyOn(
      ReactHookFormTransformer,
      'fromReactHookForm',
    );

    const onSubmitSpy = jest.fn();

    const data = [
      makeFormWithRecord(1),
      makeFormWithRecord(2),
      makeFormWithRecord(3),
    ];

    const { getByTestId } = render(
      <RecipientRegistrationScreenComponent
        mode="edit"
        data={data}
        onSubmit={onSubmitSpy}
        onCancel={() => {}}
        error={null}
        loading={false}
      />,
    );

    const submitButton = getByTestId('recipient-registration-submit-button');
    fireEvent.press(submitButton);

    await waitFor(() =>
      expect(fromReactHookFormSpy).toHaveBeenCalledWith(
        data,
        ReactHookFormTransformer.toReactHookForm(data),
      ),
    );
    expect(onSubmitSpy).toHaveBeenCalledWith(data);
  });

  it('should submit without transforming values', () => {
    const fromReactHookFormSpy = jest.spyOn(
      ReactHookFormTransformer,
      'fromReactHookForm',
    );

    const onSubmitSpy = jest.fn();

    const data = [
      makeFormWithRecord(1),
      makeFormWithRecord(2),
      makeFormWithRecord(3),
    ];

    const { getByTestId } = render(
      <RecipientRegistrationScreenComponent
        mode="review"
        data={data}
        onSubmit={onSubmitSpy}
        onCancel={() => {}}
        error={null}
        loading={false}
      />,
    );

    const submitButton = getByTestId('recipient-registration-submit-button');
    fireEvent.press(submitButton);

    expect(fromReactHookFormSpy).not.toHaveBeenCalled();
    expect(onSubmitSpy).toHaveBeenCalledWith(data);
  });
});

it('should call the onCancel callback', () => {
  const onCancelSpy = jest.fn();

  const data = [
    makeFormWithRecord(1),
    makeFormWithRecord(2),
    makeFormWithRecord(3),
  ];

  const { getByTestId } = render(
    <RecipientRegistrationScreenComponent
      mode="edit"
      data={data}
      onSubmit={() => {}}
      onCancel={onCancelSpy}
      error={null}
      loading={false}
    />,
  );

  const cancelButton = getByTestId('recipient-registration-cancel-button');
  fireEvent.press(cancelButton);

  expect(onCancelSpy).toHaveBeenCalled();
});
