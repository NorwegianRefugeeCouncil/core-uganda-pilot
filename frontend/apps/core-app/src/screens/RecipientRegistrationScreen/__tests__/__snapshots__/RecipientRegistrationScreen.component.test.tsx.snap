// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot edit mode 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RCTScrollView
    contentContainerStyle={
      Array [
        Object {},
        Object {
          "dataSet": Object {},
        },
      ]
    }
    dataSet={Object {}}
    style={
      Object {
        "marginLeft": "auto",
        "marginRight": "auto",
        "maxWidth": 1180,
        "width": "100%",
      }
    }
  >
    <View>
      <View
        dataSet={Object {}}
        style={Object {}}
      >
        <View>
          <Text>
            [{"form":{"id":"form-id-1","code":"form-code","databaseId":"database-id","folderId":"folder-id","name":"form-name-1","formType":"default","fields":[{"id":"field-id-1","name":"field-name-1","code":"","description":"","required":false,"key":false,"fieldType":{"text":{}}}]},"record":{"id":"","databaseId":"database-id","formId":"form-id-1","values":[{"fieldId":"field-id-1","value":""}]}},{"form":{"id":"form-id-2","code":"form-code","databaseId":"database-id","folderId":"folder-id","name":"form-name-2","formType":"default","fields":[{"id":"field-id-2","name":"field-name-2","code":"","description":"","required":false,"key":false,"fieldType":{"text":{}}}]},"record":{"id":"","databaseId":"database-id","formId":"form-id-2","values":[{"fieldId":"field-id-2","value":""}]}},{"form":{"id":"form-id-3","code":"form-code","databaseId":"database-id","folderId":"folder-id","name":"form-name-3","formType":"default","fields":[{"id":"field-id-3","name":"field-name-3","code":"","description":"","required":false,"key":false,"fieldType":{"text":{}}}]},"record":{"id":"","databaseId":"database-id","formId":"form-id-3","values":[{"fieldId":"field-id-3","value":""}]}}]
          </Text>
        </View>
        <View
          dataSet={Object {}}
          style={
            Object {
              "height": 16,
            }
          }
        />
        <View
          dataSet={Object {}}
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "flex-end",
            }
          }
        >
          <View
            _text={
              Object {
                "color": "darkText",
              }
            }
            accessibilityRole="button"
            accessible={true}
            collapsable={false}
            colorScheme="secondary"
            dataSet={Object {}}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onHoverIn={[Function]}
            onHoverOut={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#FFFFFF",
                "borderColor": "#24303E",
                "borderRadius": 3,
                "borderStyle": "solid",
                "borderWidth": 1,
                "flexDirection": "row",
                "height": 50,
                "justifyContent": "center",
                "paddingBottom": 8,
                "paddingLeft": 50,
                "paddingRight": 50,
                "paddingTop": 8,
              }
            }
            testID="recipient-registration-cancel-button"
          >
            <View
              dataSet={Object {}}
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                }
              }
            >
              <View
                dataSet={Object {}}
                style={Object {}}
              >
                <Text
                  dataSet={Object {}}
                  level="1"
                  style={
                    Object {
                      "backgroundColor": undefined,
                      "color": "#24303E",
                      "fontFamily": "Roboto_500Medium",
                      "fontSize": 20,
                      "fontStyle": undefined,
                      "fontWeight": undefined,
                      "letterSpacing": 0,
                      "lineHeight": 24,
                      "textAlign": "auto",
                      "textDecorationLine": undefined,
                    }
                  }
                >
                  Cancel
                </Text>
              </View>
            </View>
          </View>
          <View
            dataSet={Object {}}
            style={
              Object {
                "width": 16,
              }
            }
          />
          <View
            _text={
              Object {
                "color": "lightText",
              }
            }
            accessibilityRole="button"
            accessible={true}
            collapsable={false}
            colorScheme="primary"
            dataSet={Object {}}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onHoverIn={[Function]}
            onHoverOut={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#FF7602",
                "borderRadius": 3,
                "flexDirection": "row",
                "height": 50,
                "justifyContent": "center",
                "paddingBottom": 8,
                "paddingLeft": 50,
                "paddingRight": 50,
                "paddingTop": 8,
              }
            }
            testID="recipient-registration-submit-button"
          >
            <View
              dataSet={Object {}}
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                }
              }
            >
              <View
                dataSet={Object {}}
                style={Object {}}
              >
                <Text
                  dataSet={Object {}}
                  level="1"
                  style={
                    Object {
                      "backgroundColor": undefined,
                      "color": "#FFFFFF",
                      "fontFamily": "Roboto_500Medium",
                      "fontSize": 20,
                      "fontStyle": undefined,
                      "fontWeight": undefined,
                      "letterSpacing": 0,
                      "lineHeight": 24,
                      "textAlign": "auto",
                      "textDecorationLine": undefined,
                    }
                  }
                >
                  Review
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RNCSafeAreaProvider>
`;

exports[`should match the snapshot error 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
/>
`;

exports[`should match the snapshot loading 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
/>
`;

exports[`should match the snapshot review mode 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RCTScrollView
    contentContainerStyle={
      Array [
        Object {},
        Object {
          "dataSet": Object {},
        },
      ]
    }
    dataSet={Object {}}
    style={
      Object {
        "marginLeft": "auto",
        "marginRight": "auto",
        "maxWidth": 1180,
        "width": "100%",
      }
    }
  >
    <View>
      <View
        dataSet={Object {}}
        style={Object {}}
      >
        <View>
          <Text>
            [{"form":{"id":"form-id-1","code":"form-code","databaseId":"database-id","folderId":"folder-id","name":"form-name-1","formType":"default","fields":[{"id":"field-id-1","name":"field-name-1","code":"","description":"","required":false,"key":false,"fieldType":{"text":{}}}]},"record":{"id":"","databaseId":"database-id","formId":"form-id-1","values":[{"fieldId":"field-id-1","value":""}]}},{"form":{"id":"form-id-2","code":"form-code","databaseId":"database-id","folderId":"folder-id","name":"form-name-2","formType":"default","fields":[{"id":"field-id-2","name":"field-name-2","code":"","description":"","required":false,"key":false,"fieldType":{"text":{}}}]},"record":{"id":"","databaseId":"database-id","formId":"form-id-2","values":[{"fieldId":"field-id-2","value":""}]}},{"form":{"id":"form-id-3","code":"form-code","databaseId":"database-id","folderId":"folder-id","name":"form-name-3","formType":"default","fields":[{"id":"field-id-3","name":"field-name-3","code":"","description":"","required":false,"key":false,"fieldType":{"text":{}}}]},"record":{"id":"","databaseId":"database-id","formId":"form-id-3","values":[{"fieldId":"field-id-3","value":""}]}}]
          </Text>
        </View>
        <View
          dataSet={Object {}}
          style={
            Object {
              "height": 16,
            }
          }
        />
        <View
          dataSet={Object {}}
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "flex-end",
            }
          }
        >
          <View
            _text={
              Object {
                "color": "darkText",
              }
            }
            accessibilityRole="button"
            accessible={true}
            collapsable={false}
            colorScheme="secondary"
            dataSet={Object {}}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onHoverIn={[Function]}
            onHoverOut={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#FFFFFF",
                "borderColor": "#24303E",
                "borderRadius": 3,
                "borderStyle": "solid",
                "borderWidth": 1,
                "flexDirection": "row",
                "height": 50,
                "justifyContent": "center",
                "paddingBottom": 8,
                "paddingLeft": 50,
                "paddingRight": 50,
                "paddingTop": 8,
              }
            }
            testID="recipient-registration-cancel-button"
          >
            <View
              dataSet={Object {}}
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                }
              }
            >
              <View
                dataSet={Object {}}
                style={Object {}}
              >
                <Text
                  dataSet={Object {}}
                  level="1"
                  style={
                    Object {
                      "backgroundColor": undefined,
                      "color": "#24303E",
                      "fontFamily": "Roboto_500Medium",
                      "fontSize": 20,
                      "fontStyle": undefined,
                      "fontWeight": undefined,
                      "letterSpacing": 0,
                      "lineHeight": 24,
                      "textAlign": "auto",
                      "textDecorationLine": undefined,
                    }
                  }
                >
                  Back
                </Text>
              </View>
            </View>
          </View>
          <View
            dataSet={Object {}}
            style={
              Object {
                "width": 16,
              }
            }
          />
          <View
            _text={
              Object {
                "color": "lightText",
              }
            }
            accessibilityRole="button"
            accessible={true}
            collapsable={false}
            colorScheme="primary"
            dataSet={Object {}}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onHoverIn={[Function]}
            onHoverOut={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#FF7602",
                "borderRadius": 3,
                "flexDirection": "row",
                "height": 50,
                "justifyContent": "center",
                "paddingBottom": 8,
                "paddingLeft": 50,
                "paddingRight": 50,
                "paddingTop": 8,
              }
            }
            testID="recipient-registration-submit-button"
          >
            <View
              dataSet={Object {}}
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                }
              }
            >
              <View
                dataSet={Object {}}
                style={Object {}}
              >
                <Text
                  dataSet={Object {}}
                  level="1"
                  style={
                    Object {
                      "backgroundColor": undefined,
                      "color": "#FFFFFF",
                      "fontFamily": "Roboto_500Medium",
                      "fontSize": 20,
                      "fontStyle": undefined,
                      "fontWeight": undefined,
                      "letterSpacing": 0,
                      "lineHeight": 24,
                      "textAlign": "auto",
                      "textDecorationLine": undefined,
                    }
                  }
                >
                  Save
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RNCSafeAreaProvider>
`;
